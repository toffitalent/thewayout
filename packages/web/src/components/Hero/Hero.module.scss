@use '@styles/mixins' as *;
@use '@styles/theme';

.hero {
  min-height: 37.5rem;
  display: flex;
  align-items: center;
  padding: 5rem 0;
  position: relative;
  background-color: var(--color-grey-700);
  color: var(--color-white);

  @include mq(md) {
    min-height: 44rem;
  }

  @include mq(lg) {
    min-height: 71rem;
  }

  h1 {
    font-size: 4rem;
  }
}

.bg {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 1;
  overflow: hidden;

  &::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: 1;
    background: rgba(0, 0, 0, 0.6);
    z-index: 2;
  }
}

.container {
  @include container();
  position: relative;
  z-index: 5;

  > div {
    height: 100%;
  }
}

.actions {
  display: flex;
  justify-content: center;
  padding-top: 2rem;

  > a + a {
    margin-left: 1rem;
  }
}

