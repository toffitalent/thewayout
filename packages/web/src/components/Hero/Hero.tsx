import {
  Button,
  ButtonBase,
  ButtonLinkProps,
  ButtonProps,
  classNames,
  Flex,
  Text,
} from '@disruptive-labs/ui';
import type { StaticImageData } from 'next/image';
import Image from 'next/legacy/image';
import Link from 'next/link';
import React from 'react';
import googleImage from '@app/assets/images/google.png';
import styles from './Hero.module.scss';

export interface HeroAction extends Omit<ButtonLinkProps, 'children'> {
  href?: string;
  label: string;
}

export interface HeroProps {
  actions?: HeroAction[];
  className?: string;
  heading: string;
  image?: StaticImageData;
  subheading?: string;
  children?: React.ReactNode;
}

export const Hero: React.FC<HeroProps> = ({
  actions,
  children,
  className,
  heading,
  image,
  subheading,
}) => (
  <section className={classNames(styles.hero, className)}>
    {image && (
      <div className={styles.bg} aria-hidden="true">
        <Image src={image} alt="" layout="fill" objectFit="cover" priority />s
      </div>
    )}
    <div className={styles.container}>
      <Flex container justify="center" textAlign="center">
        <Flex item xs={12} md={8}>
          <Flex direction="row" alignItems="center" justify="center" mb={5}>
            <Text mr={1} fontWeight="700">
              Funded by
            </Text>
            <Image src={googleImage} alt="google" height={24} width={70} />
          </Flex>
          <h1>{heading}</h1>
          {subheading && <Text>{subheading}</Text>}
          {actions && (
            <div className={styles.actions}>
              {actions.map(({ href, label, ...buttonProps }) =>
                href ? (
                  <ButtonBase
                    as={Link}
                    href={href}
                    key={label}
                    colorScheme="primary"
                    {...buttonProps}
                  >
                    {label}
                  </ButtonBase>
                ) : (
                  <Button key={label} colorScheme="primary" {...(buttonProps as ButtonProps)}>
                    {label}
                  </Button>
                ),
              )}
            </div>
          )}
          {children}
        </Flex>
      </Flex>
    </div>
  </section>
);
