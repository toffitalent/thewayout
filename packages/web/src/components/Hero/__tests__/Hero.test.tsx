import React from 'react';
import { render } from '@test';
import { Hero } from '../Hero';

describe('Hero', () => {
  test('renders correctly', () => {
    const { container } = render(<Hero heading="Heading" subheading="Subheading" />);
    expect(container).toMatchSnapshot();
  });

  test('renders actions', () => {
    const { container } = render(
      <Hero heading="Heading" subheading="Subheading" actions={[{ href: '/', label: 'Action' }]} />,
    );

    expect(container).toMatchSnapshot();
  });

  test('renders image', () => {
    const { container } = render(
      <Hero heading="Heading" subheading="Subheading" image={'/test-image' as any} />,
    );

    expect(container).toMatchSnapshot();
  });
});
