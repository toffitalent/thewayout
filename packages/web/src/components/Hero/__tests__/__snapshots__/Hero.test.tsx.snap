// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders actions 1`] = `
<div>
  <section
    class="hero"
  >
    <div
      class="container"
    >
      <div
        class="container justifyContent-center textAlign-center"
      >
        <div
          class="item bp-xs-12 bp-md-8"
        >
          <div
            class="alignItems-center flexDirection-row justifyContent-center marginBottom-5"
          >
            <p
              class="text marginRight-1 fontWeight-700"
            >
              Funded by
            </p>
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2770%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="google"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <h1>
            Heading
          </h1>
          <p
            class="text"
          >
            Subheading
          </p>
          <div
            class="actions"
          >
            <a
              class="button md solid c-primary"
              href="/"
            >
              Action
            </a>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`Hero renders correctly 1`] = `
<div>
  <section
    class="hero"
  >
    <div
      class="container"
    >
      <div
        class="container justifyContent-center textAlign-center"
      >
        <div
          class="item bp-xs-12 bp-md-8"
        >
          <div
            class="alignItems-center flexDirection-row justifyContent-center marginBottom-5"
          >
            <p
              class="text marginRight-1 fontWeight-700"
            >
              Funded by
            </p>
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2770%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="google"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <h1>
            Heading
          </h1>
          <p
            class="text"
          >
            Subheading
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`Hero renders image 1`] = `
<div>
  <section
    class="hero"
  >
    <div
      aria-hidden="true"
      class="bg"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt=""
          data-nimg="fill"
          decoding="async"
          sizes="100vw"
          src="/_next/image?url=%2Ftest-image&w=3840&q=75"
          srcset="/_next/image?url=%2Ftest-image&w=640&q=75 640w, /_next/image?url=%2Ftest-image&w=750&q=75 750w, /_next/image?url=%2Ftest-image&w=828&q=75 828w, /_next/image?url=%2Ftest-image&w=1080&q=75 1080w, /_next/image?url=%2Ftest-image&w=1200&q=75 1200w, /_next/image?url=%2Ftest-image&w=1920&q=75 1920w, /_next/image?url=%2Ftest-image&w=2048&q=75 2048w, /_next/image?url=%2Ftest-image&w=3840&q=75 3840w"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
        />
      </span>
      s
    </div>
    <div
      class="container"
    >
      <div
        class="container justifyContent-center textAlign-center"
      >
        <div
          class="item bp-xs-12 bp-md-8"
        >
          <div
            class="alignItems-center flexDirection-row justifyContent-center marginBottom-5"
          >
            <p
              class="text marginRight-1 fontWeight-700"
            >
              Funded by
            </p>
            <span
              style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
            >
              <span
                style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2770%27%20height=%2724%27/%3e"
                  style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                />
              </span>
              <img
                alt="google"
                data-nimg="intrinsic"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
              <noscript />
            </span>
          </div>
          <h1>
            Heading
          </h1>
          <p
            class="text"
          >
            Subheading
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;
